### ISSUE TITLE
Issue with dataloading

### ISSUE THREAD

**Issue by tomMoral** [DataCamp-AJMMNP/dataCamp_project#1]
Hello,

I am grading your challenge but I have some issues with dataloading.
When running the `ramp-test`, I obtain:
```
$ ramp-test --quick-test
Testing Energetic class prediction for housing in Paris
Reading train and test files from ./data/ ...
Traceback (most recent call last):
  File "/home/tom/.local/miniconda/bin/ramp-test", line 8, in <module>
    sys.exit(start())
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/rampwf/utils/cli/testing.py", line 117, in start
    main()
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/rampwf/utils/cli/testing.py", line 102, in main
    assert_submission(ramp_kit_dir=ramp_kit_dir,
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/rampwf/utils/testing.py", line 105, in assert_submission
    X_train, y_train, X_test, y_test = assert_data(
  File "/home/tom/.local/miniconda/lib/python3.10/site-packages/rampwf/utils/testing.py", line 51, in assert_data
    X_test, y_test = problem.get_test_data(path=ramp_data_dir, **kwargs)
  File "/home/tom/Work/teaching/23-datacamp2023/projects/team-10-hospital-mortality/team-12-dpe/./problem.py", line 80, in get_test_data
    return _read_data(path, f_name)
  File "/home/tom/Work/teaching/23-datacamp2023/projects/team-10-hospital-mortality/team-12-dpe/./problem.py", line 57, in _read_data
    X_df['date_visite_diagnostiqueur'] = X_df['date_visite_diagnostiqueur'].astype('datetime64[ns]')

   [...]

  File "strptime.pyx", line 534, in pandas._libs.tslibs.strptime.array_strptime
  File "strptime.pyx", line 337, in pandas._libs.tslibs.strptime.array_strptime
  File "np_datetime.pyx", line 231, in pandas._libs.tslibs.np_datetime.check_dts_bounds
pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 1120-09-05 00:00:00, at position 33502. You might want to try:
    - passing `format` if your strings have a consistent format;
    - passing `format='ISO8601'` if your strings are all ISO8601 but not necessarily in exactly the same format;
    - passing `format='mixed'`, and the format will be inferred for each element individually. You might want to use `dayfirst` alongside this.
```

This is probably due to a mismatch in pandas version. Could you tell me with which version did you run this code?
If it is older than 2.1, could you fix this? This would be super helpful to be able to grade the project.
Thanks

**Comment by tomMoral** [comment]
The issue is simply that some data are corrupted (time stamp being `1120-09-03 ...` or `0`, and pandas is less permissive now with this), so this should be an easy fix. Let me know if you have trouble with this.

**Comment by JulAgu** [comment]
Hello,
I apologize for the delay in replying. I was in Holidays during the last 3 weeks.
The panda's version is **2.2.0**. I will let you know when the correction is made.

**Comment by JulAgu** [comment]
Hi,

I apologize again for not being able to solve this problem until today, only this afternoon I had time to review the code after finish work.

The code is working as planned in the last commit, the problems came from the changes made in commits a836bb670c17225165fcb361e1bc1de8df64be17 and 4031043fadb6aadf8b267672dc5c839feed67cba, in which a comment indicating a list of columns, which contained interesting information, but whose treatment required processing the text strings, was confused. The goal was to save these columns without assigning any type to them, in order to leave open the opportunity for the contestants to face real data that eventually contained many formatting problems.

Within these columns there were problems of all kinds, clearly irreconcilable using only the type transformation:

- The facade area attribute, for example, contained text and numbers indiscriminately, even changing the decimal indicator from "," to "." randomly.

- Several date columns had random comments or corrupted date strings coming from the data fill.

We found it convenient to keep these columns, however we decided to leave them available through problem.py without using them in the base line model, as this involved extensive text processing work that was beyond the scope of the project and that we found more interesting for the challenge.

**Anyway, in the last commit, I simply recovered the previous files for [problem.py](https://github.com/DataCamp-AJMMNP/dataCamp_project/blob/main/problem.py) and the introduction notebook [dpe_paris_starting_kit.ipynb](https://github.com/DataCamp-AJMMNP/dataCamp_project/blob/main/dpe_paris_starting_kit.ipynb). It was all about a communication error and that in the rush of the delivery and the end of the semester, the author of the changes did not retest the code.**

I will resend this same message by mail. As soon as you let me know that everything works fine on your machine, I will close the issue discussion.

Again an apology,
Regards,
Julian
